packages/connect-form/src/components/saved-connection-color-picker.tsx

```diff
@@ -0,0 +1,146 @@
+import React from 'react';
+import { css, cx, spacing, uiColors } from '@mongodb-js/compass-components';
+
+/**
+ * Default colors.
+ */
+const COLORS = [
+  '#5fc86e',
+  '#326fde',
+  '#deb342',
+  '#d4366e',
+  '#59c1e2',
+  '#2c5f4a',
+  '#d66531',
+  '#773819',
+  '#3b8196',
+  '#ababab',
+];
+
+const colorOptionStyles = css({
+  outline: 'none',
+  margin: 0,
+  padding: 0,
+  marginRight: spacing[2],
+  borderRadius: '50%',
+  verticalAlign: 'middle',
+  width: 36,
+  height: 36,
+  border: '1px solid transparent',
+  boxShadow: `0 0 0 0 ${uiColors.focus}`,
+  transition: 'box-shadow .16s ease-in',
+  position: 'relative',
+  overflow: 'hidden',
+});
+
+const activeColorOptionStyles = css({
+  boxShadow: `0 0 0 3px ${uiColors.focus}`,
+  transitionTimingFunction: 'ease-out',
+});
+
+const inActiveColorOptionStyles = css({
+  '&:focus, &:hover': {
+    boxShadow: `0 0 0 3px ${uiColors.gray.light1}`,
+  },
+});
+
+const noColorRedBarStyles = css({
+  width: 40,
+  borderTop: '3px solid red',
+  transform: 'rotate(-45deg)',
+  position: 'absolute',
+  left: -5,
+});
+
+const selectedColorCheckmarkStyles = css({
+  margin: 0,
+  padding: 0,
+});
+
+function ColorOption({
+  isSelected,
+  onClick,
+  hex,
+}: {
+  isSelected: boolean;
+  onClick: () => void;
+  hex: string;
+}): React.ReactElement {
+  return (
+    <button
+      style={{ background: hex }}
+      className={cx({
+        [colorOptionStyles]: true,
+        [activeColorOptionStyles]: isSelected,
+        [inActiveColorOptionStyles]: !isSelected,
+      })}
+      data-testid={`color-pick-${hex}${isSelected ? '-selected' : ''}`}
+      onClick={onClick}
+      title={hex}
+    >
```

Comment:
This should have at least `aria-pressed` to indicate that this is a toggle, not just a button. Best case this should probably be a radio group, not just disjointed buttons

Reply:
Added an `aria-pressed` to them 👍 
Making them input `type="radio"` buttons requires some overriding of html radio styles of radio boxes which I find is nice to avoid if possible. Totally down to do it but I'm not sure its worth doing here.

Reply:
I'll not push for it, whatever you prefer. My 2c are it's important to make sure we are prioritizing correctly what's the important thing here: make the behavior of this radio group align with browser defaults / accessibility standards or avoid styling the radio element (which FWIW is a very common thing to do, see leafygreen radio group component for example)